pkg/controller/common/util.go

```diff
@@ -434,38 +437,97 @@ func GetVolumeMode(pvc *corev1.PersistentVolumeClaim) corev1.PersistentVolumeMod
 	return util.ResolveVolumeMode(pvc.Spec.VolumeMode)
 }
 
-// GetStorageClassByName looks up the storage class based on the name. If no storage class is found returns nil
-func GetStorageClassByName(ctx context.Context, client client.Client, name *string) (*storagev1.StorageClass, error) {
+// GetStorageClassByNameWithK8sFallback looks up the storage class based on the name
+// If name is nil, it looks for the default k8s storage class storageclass.kubernetes.io/is-default-class
+// If no storage class is found, returns nil
+func GetStorageClassByNameWithK8sFallback(ctx context.Context, client client.Client, name *string) (*storagev1.StorageClass, error) {
+	if name == nil {
```

Comment:
Can't you just call `GetStorageClassByNameWithVirtFallback(ctx, client, name, cdiv1.DataVolumeArchive)` or if you think that is confusing, have a 3rd function that is called from these two. The amount of duplicated code is like 95% between these two functions.

Reply:
Yep definitely, thanks, pushed a change